content/en/docs/guides/build-deploy-software-via-helm-charts-with-ocm-101.md

```diff
@@ -0,0 +1,629 @@
+description: ""
+lead: ""
+date: 2024-03-19T10:36:48+01:00
+lastmod: 2024-03-19T10:36:48+01:00
+draft: false
+images: []
+menu:
+  docs:
+    parent: ""
+    identifier: "build-deploy-software-via-helm-charts-with-ocm"
+weight: 101
+toc: true
+---
+
+## Introduction
+
+Let's illustrate a very simple "Hello World" example application and show how to leverage OCM to build and deploy a Helm Chart and OCI Image on a local `kind` k8s cluster.
+The topic of `ocm` [`localization`](https://ocm.software/docs/guides/complex-component-structure-deployment/#localization) and [`configuration`](https://ocm.software/docs/guides/complex-component-structure-deployment/#configuration) is NOT part of this very simple example and is part of other in deepth guides.
+
+As base we use the `podinfo` application.
+All files can be found [here](https://github.com/stb1337/ocm-hello-world-v1).
+
+At the end of the tutorial the application will consist of the following components:
+
+* `podinfo` as our business application
+
+Following the OCM guidelines we will end-up with two component versions and six OCM OCI images.
+
+For building composed components the ["All-in-one"](../getting-started-with-ocm#all-in-one)
+mechanism becomes handy.
+
+### Requirements
+
+* [OCM command line tool](https://github.com/open-component-model/ocm)
+* [kubectl](https://kubernetes.io/docs/reference/kubectl/)
+* [git](https://git-scm.com/downloads)
+* [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+* [flux](https://fluxcd.io/flux/installation/#install-the-flux-cli)
+
+## Building OCM Software Artifact
+
+First we build an OCM component which contains Helm Charts in different kind of formats. This 101 guide explains all possible formats,
+but in reality you'll just pick the one most appropiate to you.
+
+### Prepare Helm Charts
+
+We are leveraging Kubernetes deployments which often use helm charts. The OCM specification supports helm charts as
+an artifact type. For this simple example, we will re-use already existing open source community helm charts.
+
+The OCM CLI supports referencing helm charts being stored in an OCI registry. However most
+publicly available helm charts currently are available from helm chart repositories and
+not from OCI registries. Therefore helm charts can be embedded in the `component archive` in three different ways.
+
+Reference a Helm Chart from:
+
+1. a public Helm Chart repository
+2. local `*.tgz` file
+3. local folder containing a Helm Chart file
+
+To demonstrate No. 2. and 3. we need a Helm Chart on our local machine. For the sake of the this simplified guide, we download and unpack a already existing open source Helm Chart for `podinfo`. In a real world application, this would be your own Helm Chart. You will most likely already store your own Helm Charts within a `git` repository and leverage CI/CD Pipeline to build `*.tgz` Helm Chart files in order to push to your Helm Chart Repository.
```

Comment:
```suggestion
To demonstrate No. 2. and 3. we need a Helm chart on our local machine. For the sake of the this simplified guide, we download and unpack an already existing open source Helm chart for `podinfo`. In a real world application, this would be your own Helm chart. You will most likely already store your own Helm charts within a `git` repository and leverage a CI/CD pipeline to build `*.tgz` Helm chart files in order to push to your Helm chart repository.
```